app/models/user.rb

```diff
@@ -5,8 +5,12 @@ class User < ActiveRecord::Base
          :recoverable, :rememberable, :trackable, :validatable, :confirmable
          
   enum role: [:standard, :premium, :admin]
+  
+  #relate private wikis with users through collaborators.
          
   has_many :wikis
```

Comment:
We can set up a has many through association between wiki and user by doing this:

``` ruby
#user.rb
has_many :collaborators
has_many :wikis, through: :collaborators

#wiki.rb
has_many :collaborators
has_many :users, through: collaborators
```

A collaborator object holds a wiki_id and user_id. It represents the fact that the user with the user_id is a collaborator on the wiki with the wiki_id in the collaborator object. If we have a wiki with an id of 1 and a user with an id of 2, we can make that user a collaborator by creating a collaborator object with both ids like `Collaborator.create(user_id: 2, wiki_id: 1)`. Now, because of our association in the wiki model, we can do `wiki.users` which will go through the collaborators table looking for collaborator objects where the wiki_id is 1, then grab and return the associated users. We could also do the same going from users.